adserver/views.py

```diff
@@ -2261,6 +2261,107 @@ def get_context_data(self, **kwargs):  # pylint: disable=too-many-locals
         return context
 
 
+class PublisherAuthorizedUsersView(PublisherAccessMixin, UserPassesTestMixin, ListView):
+
+    """Authorized users for a publisher."""
+
+    context_object_name = "users"
+    model = get_user_model()
+    template_name = "adserver/publisher/users.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context.update({"publisher": self.publisher})
+        return context
+
+    def get_queryset(self):
+        self.publisher = get_object_or_404(
+            Publisher, slug=self.kwargs.get("publisher_slug", "")
+        )
+        return self.publisher.user_set.all()
+
+
+class PublisherAuthorizedUsersInviteView(
+    PublisherAccessMixin, UserPassesTestMixin, CreateView
+):
+
+    """Invite additional authorized users for a publisher."""
+
+    form_class = InviteUserForm
+    model = get_user_model()
+    template_name = "adserver/publisher/users-invite.html"
+
+    def dispatch(self, request, *args, **kwargs):
+        self.publisher = get_object_or_404(
+            Publisher, slug=self.kwargs.get("publisher_slug", "")
+        )
+        return super().dispatch(request, *args, **kwargs)
+
+    def form_valid(self, form):
+        result = super().form_valid(form)
+        self.object.publishers.add(self.publisher)
+        messages.success(
+            self.request,
+            _("Successfully invited %(user)s to %(publisher)s")
+            % {"user": self.object.email, "publisher": self.publisher},
+        )
+        return result
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context.update({"publisher": self.publisher})
+        return context
+
+    def get_success_url(self):
+        return reverse(
+            "publisher_users", kwargs={"publisher_slug": self.publisher.slug}
+        )
+
+
+class PublisherAuthorizedUsersRemoveView(
+    PublisherAccessMixin, UserPassesTestMixin, TemplateView
+):
+
+    """
+    Remove authorized users for a publisher.
+
+    This doesn't remove or deactivate the user - just removes them from the publisher.
```

Comment:
If they login, they will just be shown a screen with no publishers/advertisers right?

Reply:
That is correct. They get a a message "You don't have access to anything": https://github.com/readthedocs/ethical-ad-server/blob/5f7dc3b30510f80266f99bf261abbbfbe85b2348/adserver/templates/adserver/dashboard.html#L58